pkg/safemath/safemath.go

```diff
@@ -0,0 +1,33 @@
+package safemath
+
+import "math/bits"
+
+// This is a constant-time version of the following function:
```

Comment:
It's become a part of jargon to call such functions constant-time because they're literally constant-time not asymptotically. On every input they perform the same amount of low-level instructions. This package is an example https://pkg.go.dev/crypto/subtle